test/infrastructure/docker/README.md

```diff
@@ -11,3 +11,7 @@ For a complete overview, please refer to the documentation available [here](http
 * To be a the reference implementation of an infrastructure provider.
 * The code is highly trusted and used in testing of ClusterAPI.
 * This provider can be used as a guide for developers looking to implement their own infrastructure provider.
+
+## End-to-end testing
+
+In the top level directory, cluster-api, run `make -C test/infrastructure/docker test-e2e` to run the test suite locally.
```

Comment:
Can we add instruction to show how to run the e2e test with your changes, like build an image and replace it in _some_yaml_file_

Reply:
This will test local changes by building a local image and loading it into the kind cluster!

But I think the feedback I'm reading from this is to describe what this is testing exactly

Reply:
Yes, it was not clear to me :)

Reply:
Switched up the wording a bit, how does it look now?

Reply:
Very clear to me now, Thanks 👍